server/internal/server/status.go

```diff
@@ -59,7 +98,108 @@ func (s *ISS) Stop() {
 	s.srv.Stop()
 }
 
+// Refresh refreshes the inference status.
+func (s *ISS) Refresh(ctx context.Context, interval time.Duration) error {
```

Comment:
A bit overkill to have caching/refreshing? I think it's ok to compute whenever a request comes. If we keep, can't we make the refresh period configurable?